appservice/src/connectToGitHub.ts

```diff
@@ -78,14 +111,22 @@ async function showGitHubAuthPrompt(): Promise<void> {
     }
 }
 
-async function getJsonRequest(url: string, requestOptions: WebResource, node: IAzureNode): Promise<Object[]> {
+async function getJsonRequest(requestOptions: gitHubWebResource, node: IAzureNode): Promise<Object[]> {
     // Reference for GitHub REST routes
     // https://developer.github.com/v3/
     // Note: blank after user implies look up authorized user
     try {
         // tslint:disable-next-line:no-unsafe-any
-        const gitHubResponse: string = await requestP.get(url, <WebResource>requestOptions);
-        return <Object[]>JSON.parse(gitHubResponse);
+        const gitHubResponse: Response = await request(requestOptions).promise();
+        if (gitHubResponse.headers.link) {
+            const link: string = <string>gitHubResponse.headers.link;
+            // regex to find the next and last links in the string
+            const linkUrls: string[] = link.match(/https\:\/\/[a-zA-Z0-9\-\.]+\.[a-zA-Z]{2,3}(\/\S*)?[0-9]/g);
```

Comment:
What exactly does 'link' look like? Why do you have to use this regex? There's no easier way to get this information?

It feels like you're making assumptions about the ordering, etc. and I'm just worried about another issue like the customDomains thing

Reply:
Link is attached to the header of the response if there are additional repos.  This is what the response looks like:

"<https://api.github.com/organizations/6154722/repos?page=2>; rel="prev", <https://api.github.com/organizations/6154722/repos?page=4>; rel="next", <https://api.github.com/organizations/6154722/repos?page=59>; rel="last", <https://api.github.com/organizations/6154722/repos?page=1>; rel="first" "

For whatever reason, it comes as a string so I have to parse it like that to retrieve the urls.  That being said, it looks like you probably made me find a bug because I never paid attention to rel="prev" part before.

Reply:
Yeah I would leverage the next/prev/last things somehow rather than making an assumption about the order. Also if you can be less strict about the url regex I think that might be safer. It feels like the format is always `<url>; rel="<rel>"`. For example - maybe you split on "," and then use a regex like `(.*);[\s]*rel="(.*)"`